docs/core_docs/docs/integrations/tools/goat.mdx

```diff
@@ -0,0 +1,99 @@
+---
+hide_table_of_contents: true
+---
+
+# GOAT
+
+[GOAT](https://github.com/goat-sdk/goat) is the finance toolkit for AI agents.
+
+## Overview
+
+Create agents that can:
+
+- Send and receive payments
+- Purchase physical and digital goods and services
+- Engage in various investment strategies:
+  - Earn yield
+  - Bet on prediction markets
+- Purchase crypto assets
+- Tokenize any asset
+- Get financial insights
+
+### How it works
+GOAT leverages blockchains, cryptocurrencies (such as stablecoins), and wallets as the infrastructure to enable agents to become economic actors:
+
+1. Give your agent a [wallet](https://github.com/goat-sdk/goat/tree/main#chains-and-wallets)
+2. Allow it to transact [anywhere](https://github.com/goat-sdk/goat/tree/main#chains-and-wallets)
+3. Use more than [+200 tools](https://github.com/goat-sdk/goat/tree/main#tools)
+
+See everything GOAT supports [here](https://github.com/goat-sdk/goat/tree/main#chains-and-wallets).
+
+**Lightweight and extendable**
+Different from other toolkits, GOAT is designed to be lightweight and extendable by keeping its core minimal and allowing you to install only the tools you need.
+
+If you don't find what you need on our more than 200 integrations you can easily:
+
+- Create your own plugin
+- Integrate a new chain
+- Integrate a new wallet
+- Integrate a new agent framework
+
+See how to do it [here](https://github.com/goat-sdk/goat/tree/main#-contributing).
+
+
+## Setup
+
+1. Install the core package and langchain adapter:
+
+```bash
+pip install @goat-sdk/core @goat-sdk/adapter-langchain
+```
+
+2. Install the type of wallet you want to use (e.g solana):
+
+```bash
+pip install @goat-sdk/wallet-evm @goat-sdk/wallet-viem
+```
+
+3. Install the plugins you want to use in that chain:
+
+```bash
+pip install @goat-sdk/plugin-erc20
+```
+
+## Instantiation
+
+Now we can instantiate our toolkit:
+
+```typescript
+import { http } from "viem";
+import { createWalletClient } from "viem";
+import { privateKeyToAccount } from "viem/accounts";
+import { baseSepolia } from "viem/chains";
+
+import { getOnChainTools } from "@goat-sdk/adapter-langchain";
+import { PEPE, USDC, erc20 } from "@goat-sdk/plugin-erc20";
+
+import { sendETH } from "@goat-sdk/wallet-evm";
+import { viem } from "@goat-sdk/wallet-viem";
+
+// 1. Create a wallet client
+const account = privateKeyToAccount(process.env.WALLET_PRIVATE_KEY as `0x${string}`);
+
+const walletClient = createWalletClient({
+    account: account,
+    transport: http(process.env.RPC_PROVIDER_URL),
+    chain: baseSepolia,
+});
+
+const tools = await getOnChainTools({
+    wallet: viem(walletClient),
+    plugins: [sendETH(), erc20({ tokens: [USDC, PEPE] })],
+});
```

Comment:
Can you add an example here that uses this with `createReactAgent` from LangGraph?

https://langchain-ai.github.io/langgraphjs/how-tos/create-react-agent/

Reply:
done! :) thanks for the review!